### ISSUE TITLE
Are you willing to provide suggestions on  [using kasini3000 in a separate alpine disk]?

### ISSUE THREAD

**Issue by kasini3000** [kasini3000/kasini3000#31]
For the security of the Linux host computer, I recommend using alpine linux for the following reasons:
* Its software package is small and safe enough.
* It takes up memory and has very little cpu.
* The Kasini 3000 can use both the Windows host and the Linux host at the same time.
* It is basically live cd, that is, linux started from iso. Just download the new iso version to complete the linux upgrade.
* It supports installation to the hard disk. After installation, you can still start through the ios on the CD, which basically does not take up hard disk space.
* You can store the/etc/kasini3000 directory on the hard disk and install the powershell on the hard disk.
* Using kasini3000 in alpine linux is no different from the other linux distribution.

I have also tried microsoft cblm 2.0, which seems to be not well. The main reason is that after installation on the hard disk, it takes up a lot of disk space, which is no different from the ordinary Linux distribution.

**Comment by rhubarb-geek-nz** [comment]
Alpine runs happily under Docker on Windows, and is a very small container image.

Ensure you get the alpine compiled version of powershell from https://github.com/PowerShell/PowerShell/releases/download/v7.2.6/powershell-7.2.6-linux-alpine-x64.tar.gz

Alpine binaries normally need a different compilation as they use a different C library for example 

Alpine...

```
$ ldd /usr/bin/pwsh
	/lib/ld-musl-x86_64.so.1 (0x7f71b0750000)
	libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x7f71b059a000)
	libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x7f71b0580000)
	libc.musl-x86_64.so.1 => /lib/ld-musl-x86_64.so.1 (0x7f71b0750000)
```

Ubuntu...

```
$ ldd /usr/bin/pwsh
	linux-vdso.so.1 (0x00007ffcad7ba000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007efd270ff000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007efd270fa000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007efd26ed0000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007efd26de9000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007efd26dc9000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007efd26b9f000)
	/lib64/ld-linux-x86-64.so.2 (0x00007efd27131000)
```

I would recommend ensuring you are using an Alpine compiled version of SQLite.Interop.dll

The current kasini3000 packaged linux.SQLite.Interop.dll contains

```
$ objdump -p linux.SQLite.Interop.dll | grep NEEDED
  NEEDED               libm.so.6
  NEEDED               libpthread.so.0
  NEEDED               libdl.so.2
  NEEDED               libc.so.6
```

**Comment by rhubarb-geek-nz** [comment]
If you build SQLite.Interop.dll on Alpine you get

```
$ objdump -p SQLite.Interop.dll | grep NEEDED
  NEEDED               libc.musl-x86_64.so.1
```

**Comment by rhubarb-geek-nz** [comment]
If you want to keep the installation on Alpine small then don't clone from git and remove all the redundant Windows binaries.

**Comment by rhubarb-geek-nz** [comment]
If you are running from a CD does this mean the root file system is readonly? Given that kasin3000 can only run master node as root then running using a readonly root file system is a great idea.

I would recommend that everything exception /root/kasini3000 is on the readonly partition, with the data on a separate mounted volume/partition.

This includes having /etc/kasini3000 and PowerShell installed on the read only partition. A better way to manage /etc/kasini3000 is to install the software in /usr/share/kasini3000 and make /etc/kasini3000 a symbolic link to it.

**Comment by rhubarb-geek-nz** [comment]
> This includes having /etc/kasini3000 and PowerShell installed on the read only partition. 

Unfortunately PowerShell can't run on a read only file system

```
Failed to create CoreCLR, HRESULT: 0x80004005
```

**Comment by kasini3000** [comment]
hi:
linux  x86-64 version of SQLite.Interop.dll work fine on alpine 3.13 x86-64 , 
tested by me!

to test:
```
#pwsh
/etc/kasini3000/cmdb/cmdb_test.ps1 
# run twice, or more. normal on 2rd, and 3rd run.
```

**Comment by rhubarb-geek-nz** [comment]
Are you using the alpine build of powershell? Eg powershell-7.3.0-linux-alpine-x64.tar.gz?

**Comment by kasini3000** [comment]
yes,
https://learn.microsoft.com/en-us/powershell/scripting/install/install-alpine?view=powershell-7.3

if use alpine 3.17 ,then  replace libssl1.1   to libssl3

**Comment by rhubarb-geek-nz** [comment]
Fair enough. The reason I mentioned it was to avoid loading conflicting shared libraries into the same process, Alpine is libc.musl based which is why PowerShell has a separate build for it.
